### ISSUE TITLE
Mist.io review

### ISSUE THREAD

**Issue by joshgav** [cncf/tag-app-delivery#214]
Presented by @cpsaltis to TAG App Delivery on 2022-05-04.

- Site: https://mist.io/
- GitHub: https://github.com/mistio/mist-ce
- Description: A common cloud resource API over custom provider (Azure, AWS, etc.) APIs.
    - For example, enables simple switching of workloads between providers for cost optimization.
- Are all operations in the underlying providers possible?
    - No, only some common denominator, like Crossplane, libcloud and other projects.
    - How will you stay in sync with provider APIs? Crossplane and Terraform have done a lot of work.
- Does it make sense in CNCF?
    - Yes cause it's a project to create an open standard for cloud resource APIs.
    - Compare to [libcloud](https://libcloud.apache.org/), [fog](https://fog.io/), [jclouds](https://jclouds.apache.org/).
    - Use libcloud (Apache) for some stuff. Consider Apache to host?
- Does it overlap with Crossplane?
    - It's the same concept of abstracting provider control planes, but it's not based on Kubernetes and CRDs - instead based on imperative APIs.

**Comment by cpsaltis** [comment]
Thanks a lot for summing these up @joshgav. 

I would like to add a few more notes.

To the staying in sync with provider APIs question:

- There is a big core of functionality already available.
- This core is mostly based on Apache Libcloud. We are getting updates from the community and we are contributing back the things we develop. Later this year we also plan to integrate with Terraform. This way our users will gain access to new functionality from there as well.

Regarding maintenance, APIs don’t change very often and when they do there are minor changes. To detect such issues quickly we have invested a lot in automated testing. 

About the overlap with Crossplane, the main differences we see are the following:

1. With Mist you can autodiscover and manage resources even if they were not created by Mist itself, e.g. in the case of brownfield deployments or using Mist in parallel with other tools. 
1. Mist introduces abstractions over resource types. For example, to Mist all VMs are machines that share common attributes. Same goes e.g. for clusters, volumes etc. There are cases where we dig deeper and provide more provider-specific functionality but overall our goal is to abstract away as much of it as possible. This helps users manage their infrastructure in a more general and friendly way.

To sum things up, Mist is targeting cases with very heterogeneous setups that combine cloud native with more traditional VM-based architectures.

**Comment by thschue** [comment]
Will close this issue as there are no further action items associated with it.
